src/Sulu/Bundle/AdminBundle/Resources/js/containers/Form/Field.js

```diff
@@ -32,10 +35,42 @@ class Field extends React.Component<Props> {
         showAllErrors: false,
     };
 
+    @computed get conditionData() {
+        const {data, formInspector} = this.props;
+        const {locale, metadataOptions, options} = formInspector;
+
+        return conditionDataProviderRegistry.getAll().reduce(
+            function(data, conditionDataProvider) {
+                return {...data, ...conditionDataProvider(data, options, metadataOptions)};
+            },
+            {...data, __locale: locale}
+        );
+    }
+
+    @computed get disabled() {
+        const {schema} = this.props;
+
+        if (!schema.disabledCondition) {
+            return false;
+        }
+
+        return jexl.evalSync(schema.disabledCondition, this.conditionData);
+    }
+
+    @computed get visible() {
+        const {schema} = this.props;
+
+        if (!schema.visibleCondition) {
+            return true;
+        }
+
+        return jexl.evalSync(schema.visibleCondition, this.conditionData);
+    }
```

Comment:
I've tried to add the evaluations in this file, because in here the two conditions can be evaluated for this exact field, without the need to work with array/objects or whatever. Since the `Field` is already an observer, only the necessary fields are reevaluated.

Reply:
But there is one problem with that approach. The [`Renderer` component](https://github.com/sulu/sulu/blob/dc4b04b77af71286cae13aad44d704a83e1755ee/src/Sulu/Bundle/AdminBundle/Resources/js/containers/Form/Renderer.js#L44) also renders the form sections, so the `visibleCondition` handling would have to be copied there. But I think I would move the rendering of `Section` in this `Field` component, to reuse that logic.

Reply:
Would be okay for me, but I am not sure if `Field` is still the correct name then

Reply:
Not really reimplement, I think it will end up looking a bit different in general, but the `renderItem` method in the `Renderer` would be gone.

Reply:
And regarding the name, what would you suggest? Initially I would have handle the section exactly the same as another field type, in order to avoid a situation like we have now :see_no_evil:

Reply:
maybe just `Item`?

Reply:
I've introduced another `Section` component in here, and duplicated the `visibleCondition` getter for now... Getting both of them to work as a single component is not that easy, because they are sharing quite some data via recursion... I'll have another look at that at the end, if there is some time left.